### PR TITLE
logsetup: enable logging for craft-providers (CRAFT-149)

### PR DESCRIPTION
**Pull request by cjp256**
Enable logging for craft-providers library, matching the current
logging behaviors for charmcraft.  Add 'enabled_loggers' list to
support doing this in an easily extensible manner.

Example log snippet when running with --verbose:
```
2021-06-17 11:24:46,538  charmcraft                     DEBUG    Starting charmcraft version 0.10.0+199.g6541d0a.dirty
2021-06-17 11:24:46,538  charmcraft.main                DEBUG    Raw pre-parsed sysargs: args={'help': False, 'verbose': True, 'quiet': False, 'project_dir': None} filtered=['pack']
2021-06-17 11:24:46,539  charmcraft.main                DEBUG    General parsed sysargs: command='pack' args=[]
2021-06-17 11:24:46,540  charmcraft.main                DEBUG    Command parsed sysargs: Namespace(entrypoint=None, requirement=None, debug=False, destructive_mode=False, shell=False)
2021-06-17 11:24:46,540  charmcraft.commands.pack       DEBUG    working arguments: {'from': PosixPath('/home/chris/git/charmx/bases'), 'entrypoint': PosixPath('/home/chris/git/charmx/bases/src/charm.py'), 'requirement': [PosixPath('/home/chris/git/charmx/bases/requirements.txt')], 'destructive_mode': False, 'debug': False, 'shell': False}
2021-06-17 11:24:46,540  charmcraft.providers           DEBUG    Launching build environment...
2021-06-17 11:24:46,540  craft_providers.multipass.multipass DEBUG    Executing on host: multipass list --format json
...
```

Signed-off-by: Chris Patterson <chris.patterson@canonical.com>

### CODE REVIEW

File: charmcraft/logsetup.py [hunk 1]
```diff
@@ -25,8 +25,11 @@
 FORMATTER_SIMPLE = "%(message)s"
 FORMATTER_DETAILED = "%(asctime)s  %(name)-30s %(levelname)-8s %(message)s"
 
-_logger = logging.getLogger("charmcraft")
-_logger.setLevel(logging.DEBUG)
+logger = logging.getLogger("charmcraft")
+enabled_loggers = [
+    logging.getLogger("charmcraft"),
```
Comment (facundobatista) [inline charmcraft/logsetup.py:30]: Maybe just use `logger` and not re-ask it?

**Review by facundobatista — COMMENTED** [review]
I like the code! But the final result is ugly, I'm -1 to lose "the columns".

we could increase the blank space for the "charmcraft" command. OTOH, `craft_providers.multipass.multipass` is like too long and redundant, maybe that could be improved?

**Review by cmatsuoka — APPROVED** [review]
Looks good. We'll need a similar approach in craft-parts.

File: charmcraft/logsetup.py (hunk 1 above)
Comment (cjp256) [reply charmcraft/logsetup.py:30]: As far as the names, I use __name__ with getLogger so it's not trivial to shorten it.  I can throw another tab or two in there to improve alignment though.

Comment (cjp256) [reply charmcraft/logsetup.py]: Fixed

- Added 10 more spaces which should be plenty.
- Re-used `logger` too.

**Review by facundobatista — APPROVED** [review]
Great, thanks!

**Pull request merged**